shell/platform/embedder/embedder.h

```diff
@@ -1064,6 +1064,53 @@ typedef int64_t FlutterPlatformViewIdentifier;
 FLUTTER_EXPORT
 extern const int32_t kFlutterSemanticsNodeIdBatchEnd;
 
+typedef enum {
+  // Indicates the string should be announced character by character.
+  kSpellOut,
+  // Indicates the string should be announced using the specified locale.
+  kLocale,
+} FlutterStringAttributeType;
+
+// Indicates the assistive technology should announce out the string character
+// by character.
+//
+// See dart:ui's SpellOutStringAttribute.
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterSpellOutStringAttribute).
+  size_t struct_size;
+} FlutterSpellOutStringAttribute;
+
+// Indicates the assistive technology should announce the string using the
+// specified locale.
+//
+// See dart:ui's LocaleStringAttribute.
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterLocaleStringAttribute).
+  size_t struct_size;
+  // The locale of this attribute.
+  const char* locale;
+} FlutterLocaleStringAttribute;
+
+// Indicates how the assistive technology should treat the string.
+//
+// See dart:ui's StringAttribute.
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterStringAttribute).
+  size_t struct_size;
+  // The position this attribute starts.
+  size_t start;
+  // The next position after the attribute ends.
+  size_t end;
+  /// The type of the attribute described by the subsequent union.
+  FlutterStringAttributeType type;
+  union {
+    // Indicates the string should be announced character by character.
+    const FlutterSpellOutStringAttribute* spell_out;
+    // Indicates the string should be announced using the specified locale.
+    const FlutterLocaleStringAttribute* locale;
+  };
```

Comment:
I realise this is more of a question for the dart:ui side of this change, but are these really mutually exclusive? I could imagine someone writing a language learning app to teach English to French speakers, where the French word OVNI is presented in French, then the English word UFO is spelt out. Fundamentally, these two properties seem independent of each other. For OSes/embedders where it's mutually exclusive, I could imagine some very simple resolution algorithm.

Reply:
You can apply overlapping string attributes to a string. For example, you can do something like:

```yaml
value: Hello abc
value attributes:
  - start 0
    end: 10
    type: kLocale 
    locale: en-US
  - start 6
    end: 10
    type: kSpellOut
```

Assuming I can count, this means that the entire string should be announced using the `en-US` locale but that only the last 3 characters should be announced character by character. In other words, these string attributes aren't mutually exclusive. Please let me know if I misunderstood your question.

Reply:
Ah, I see; from the user's point of view we simply apply multiple range+type attributes with overlapping ranges rather than, for example, a single range+types attribute that applies both. Makes sense; thanks. We may want to point out that use case in the framework-side documentation if it isn't already.